champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/axe/WolfsFury.java

```diff
@@ -105,31 +117,55 @@ public void onDamage(CustomDamageEvent e) {
             e.addReason(getName());
         }
         missedSwings.put(damager, 0);
-
     }
 
-    @UpdateEvent(delay = 500)
+    @UpdateEvent
     public void onUpdate() {
-        active.entrySet().removeIf(entry -> expire(entry.getKey(), false));
+        Iterator<Map.Entry<Player, Long>> iterator = active.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<Player, Long> entry = iterator.next();
+            Player player = entry.getKey();
+            if (player == null) {
+                iterator.remove();
+            } else {
+                spawnSkillParticles(player);
+                if (entry.getValue() - System.currentTimeMillis() <= 0) {
```

Comment:
Call the `expire` function instead of doing this. It also has other checks for a failsafe